drivers/gpio/gpio_rp1.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2024 Junho Lee <junho@tsnlab.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_rp1_gpio
+
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define GPIO_STATUS(base, n) (base + 0x8 * n)
+#define GPIO_CTRL(base, n)   (GPIO_STATUS(base, n) + 0x4)
+
+#define GPIO_STATUS_OUT_TO_PAD    0x200
+#define GPIO_STATUS_OUT_FROM_PERI 0x100
+
+#define GPIO_CTRL_OUTOVER_MASK 0x3000
+#define GPIO_CTRL_OUTOVER_PERI 0x0
+
+#define GPIO_CTRL_OEOVER_MASK 0xc000
+#define GPIO_CTRL_OEOVER_PERI 0x0
+
+#define GPIO_CTRL_FUNCSEL_MASK 0x001f
+#define GPIO_CTRL_FUNCSEL_RIO  0x5
+
+#define RIO_OUT(base) (base)
+#define RIO_OE(base)  (base + 0x4)
+#define RIO_IN(base)  (base + 0x8)
+
+#define RIO_SET 0x2000
+#define RIO_CLR 0x3000
+
+#define RIO_OUT_SET(base) (RIO_OUT(base) + RIO_SET)
+#define RIO_OUT_CLR(base) (RIO_OUT(base) + RIO_CLR)
+
+#define RIO_OE_SET(base) (RIO_OE(base) + RIO_SET)
+#define RIO_OE_CLR(base) (RIO_OE(base) + RIO_CLR)
+
+#define PADS_CTRL(base, n) (base + 0x4 * (n + 1))
+
+#define PADS_OUTPUT_DISABLE 0x80
+#define PADS_INPUT_ENABLE   0x40
+
+#define PADS_PULL_UP_ENABLE   0x8
+#define PADS_PULL_DOWN_ENABLE 0x4
+
+#define DEV_CFG(dev)  ((const struct gpio_rp1_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct gpio_rp1_data *)(dev)->data)
```

Comment:
These macros are not used. Please remove.

Reply:
I'm sorry for my late response. Those macros are used by DEVICE_MMIO* macros and it won't compile without them. 
https://github.com/zephyrproject-rtos/zephyr/blob/25b6cd07fe0283285e75887483c8cb2d6f4b8f14/drivers/gpio/gpio_rp1.c#L199